woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php

```diff
@@ -75,41 +77,44 @@ public function one_click_handler() {
+				'first_name' => get_user_meta($this->userId, 'billing_first_name', true),
+				'last_name' => get_user_meta($this->userId, 'billing_last_name', true),
+				'company' => get_user_meta($this->userId, 'billing_company', true),
+				'address_1' => get_user_meta($this->userId, 'billing_address_1', true),
+				'address_2' => get_user_meta($this->userId, 'billing_address_2', true),
+				'city' => get_user_meta($this->userId, 'billing_city', true),
+				'state' => get_user_meta($this->userId, 'billing_state', true),
+				'postcode' => get_user_meta($this->userId, 'billing_postcode', true),
+				'phone' => get_user_meta($this->userId, 'billing_phone', true)
 			);
 
 			$product_to_add = get_product( $product_id );
 
 			$order->add_product( $product_to_add, 1 );
-			$order->set_billing_email($user['email']);
-			$order->set_billing_country($user['country']);
-			$order->set_billing_first_name($user['first_name']);
-			$order->set_billing_last_name($user['last_name']);
-			$order->set_billing_company($user['company']);
-			$order->set_billing_address_1($user['address_1']);
-			$order->set_billing_address_2($user['address_2']);
-			$order->set_billing_city($user['city']);
-			$order->set_billing_state($user['state']);
-			$order->set_billing_phone($user['phone']);
+			
+			update_post_meta($order->id, '_billing_email', $user['email']);
+			update_post_meta($order->id, '_billing_country', $user['country']);
+			update_post_meta($order->id, '_billing_first_name', $user['first_name']);
+			update_post_meta($order->id, '_billing_last_name', $user['last_name']);
+			update_post_meta($order->id, '_billing_company', $user['company']);
+			update_post_meta($order->id, '_billing_address_1', $user['address_1']);
+			update_post_meta($order->id, '_billing_address_2', $user['address_2']);
+			update_post_meta($order->id, '_billing_city', $user['city']);
+			update_post_meta($order->id, '_billing_state', $user['state']);
+			update_post_meta($order->id, '_billing_phone', $user['phone']);
+
 			$order->set_payment_method($this->gateway);
-			$order->set_total( WC()->cart->shipping_total, 'shipping' );
-			$order->set_total( WC()->cart->get_cart_discount_total(), 'cart_discount' );
-			$order->set_total( WC()->cart->get_cart_discount_tax_total(), 'cart_discount_tax' );
-			$order->set_total( WC()->cart->tax_total, 'tax' );
-			$order->set_total( WC()->cart->shipping_tax_total, 'shipping_tax' );
-			$order->set_total( WC()->cart->total );
+			
+			update_post_meta($order->id, '_order_shipping', WC()->cart->shipping_total );
+			update_post_meta($order->id, '_cart_discount', WC()->cart->get_cart_discount_total() );
+			update_post_meta($order->id, '_cart_discount_tax', WC()->cart->get_cart_discount_tax_total() );
+			update_post_meta($order->id, '_order_tax', WC()->cart->tax_total );
+			update_post_meta($order->id, '_order_shipping_tax', WC()->cart->shipping_tax_total );
+			update_post_meta($order->id, '_order_total', WC()->cart->total );
```

Comment:
I wonder if a sequence of updates won't cause a slowdown...
This will definitely benefit from an abstraction layer.